src/components/Details/WriteReview.js

```diff
@@ -39,11 +43,34 @@ const StyledRating = styled(Rating)(({ theme }) => ({
   position: "center%",
 }));
 
-function WriteReview({albumName}) {
+function WriteReview({albumName, albumIDFromDB}) {
+
+  const { loggedInUser, loggedIn } = useSelector(state => state.loggedInUserData);
+  const [userIDFromDB, setUserIDFromDB] = useState();
+  const [userReview, setUserReview] = useState();
   const [reviewText, setReviewText] = useState('');
   const [stars, setStars] = useState(0);
+  const [shouldFindReview, setShouldFindReview] = useState(true);
+
+  async function fetchUserReviewsFromDatabase() {
+    const userFromDB = await getUser(loggedInUser.username);
+    setUserIDFromDB(userFromDB.id);
+    const allUserReviews = await getReviewsForUser(userIDFromDB);
+    const foundUserReview = allUserReviews.find(review => review.AlbumId === albumIDFromDB && review.UserId === userIDFromDB);
```

Comment:
It may be worth making a backend endpoint down the line for getting a single review to save the trouble of doing this filtering on the front end, but this works for now.